staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -1339,9 +1339,9 @@ func (c *cacheWatcher) convertToWatchEvent(event *watchCacheEvent) *watch.Event
 
 	switch {
 	case curObjPasses && !oldObjPasses:
-		return &watch.Event{Type: watch.Added, Object: getEventObject(event.Object)}
+		return &watch.Event{Type: watch.Added, Object: event.Object}
```

Comment:
make sure you comment this precisely everywhere that the object is not owned by the event.